File: 0078-subsets/0078-subsets.java
class Solution {
    List<List<Integer>> res = new LinkedList<>();
    public List<List<Integer>> subsets(int[] nums) {
        List<Integer> track = new LinkedList<>();
        backtrack(nums, 0, track);
        return res;
    }

    void backtrack(int[] nums, int start, List<Integer> track) {
        res.add(new LinkedList<>(track));
        for (int i = start; i < nums.length; i++) {
            track.add(nums[i]);
            backtrack(nums, i+1, track);
            track.remove(track.size()-1);
        }
    }
}

File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        ans = []
        path = []
        n = len(nums)
        def dfs(i):
            ans.append(path[:])
            if i == n: 
                return
            for j in range (i, n):
                path.append(nums[j])
                dfs(j+1)
                path.remove(nums[j])
        dfs(0)
        return ans